src/pin.ts

```diff
@@ -0,0 +1,436 @@
+    await mkdir(versionDir, { recursive: true })
+  }
+
+  /**
+   * Copy contract files to the version directory
+   * This includes both the main contract and its slim version if it exists
+   */
+  private async copyContractFiles (contractPath: string, contractName: string, version: string) {
+    const sourceDir = dirname(join(this.projectRoot, contractPath))
+    const targetDir = join(this.projectRoot, 'contracts', contractName, version)
+
+    // Copy main contract file
+    const mainFile = `${contractName}.js`
+    const mainSource = join(sourceDir, mainFile)
+    const mainTarget = join(targetDir, mainFile)
+
+    if (existsSync(mainSource)) {
+      await copyFile(mainSource, mainTarget)
+      console.log(colors.green(`✅ Copied ${mainFile}`))
+    } else {
+      throw new Error(`Main contract file not found: ${mainSource}`)
+    }
+
+    // Copy slim version if it exists
+    const slimFile = `${contractName}-slim.js`
+    const slimSource = join(sourceDir, slimFile)
+    const slimTarget = join(targetDir, slimFile)
+
+    if (existsSync(slimSource)) {
+      await copyFile(slimSource, slimTarget)
+      console.log(colors.green(`✅ Copied ${slimFile}`))
+    } else {
+      console.log(colors.yellow(`⚠️  Slim version not found: ${slimFile} (this is optional)`))
+    }
+
+    // Copy any existing manifest file
+    const manifestFile = `${contractName}.manifest.json`
+    const manifestSource = join(sourceDir, manifestFile)
+    const manifestTarget = join(targetDir, manifestFile)
+
+    if (existsSync(manifestSource)) {
+      await copyFile(manifestSource, manifestTarget)
+      console.log(colors.green(`✅ Copied ${manifestFile}`))
+    }
+  }
+
+  /**
+   * Update chelonia.json with the new contract version
+   * This only updates the specific contract being pinned
+   */
+  private async updateCheloniaConfig (contractName: string, version: string, contractPath: string) {
+    // Update the specific contract's configuration
+    this.cheloniaConfig.contracts[contractName] = {
+      version,
+      path: contractPath
+    }
+
+    // Write the updated configuration back to chelonia.json
+    const configPath = join(this.projectRoot, 'chelonia.json')
+    const configContent = JSON.stringify(this.cheloniaConfig, null, 2) + '\n'
+
+    await writeFile(configPath, configContent, 'utf8')
+    console.log(colors.green('✅ Updated chelonia.json'))
+    console.log(colors.gray(`Set ${contractName} to version ${version}`))
+  }
+
+  /**
+   * Generate manifest for the specific contract version
+   * This creates the .manifest.json file needed by chel deploy
+   *
+   * The manifest format must match what deploy.ts expects:
+   * - head: JSON string with manifestVersion
+   * - body: JSON string with contract details (name, version, contract.file, contractSlim.file)
+   * - signature: cryptographic signature (placeholder for now)
+   */
+  private async generateContractManifest (contractName: string, version: string) {
```

Comment:
Why are you implementing this, when `manifest.ts` already exists and actually works?